src/python/pants/backend/codegen/tasks/simple_codegen_task.py

```diff
@@ -104,6 +188,15 @@ def assert_sources_present(self, sources, targets):
                       .format(count=len(targets), targets=formatted_targets))
 
   def execute(self):
+    execute_strategies = {
+      'isolated': self._execute_strategy_isolated,
+      'global': self._execute_strategy_global,
+    }
+    sources_strategies = {
+      'isolated': self._find_sources_strictly_generated_by_target,
+      'global': self.sources_generated_by_target,
+    }
```

Comment:
Eric, you earlier said 
```
I think this might be better handled by creating a class that is like an interface
(There is an @abstractmethod decorator in python you could use, e.g. from abc import
abstractmethod ) and then create two concrete classes.
```
Do you mean a structure like:
```
interface CodegenStrategy:
  +abstract execute_codegen()
  +abstract find_sources()
```
?